packages/query-core/src/queryObserver.ts

```diff
@@ -82,6 +85,13 @@ export class QueryObserver<
 
     this.#client = client
     this.#selectError = null
+    this.#currentThenable = pendingThenable()
```

Comment:
I just want to double check that I'm thinking about this correctly. This is a "proxy"-thenable that is kept in sync with the promise that lives in the query cache, and the mechanism for that "syncing" is the `updateResult`-function?

So any means of resolving, rejecting or replacing the original promise needs to call `updateResult` on all observers to "sync" to the thenables?

Just thinking out loud, this makes me wonder what happens if you:

* useQuery -> use(promise)
* fetchQuery / refetch before above query resolves, and either:
  * The first query resolves first
  * The fetchQuery / refetch resolves first

Does this have a defined behaviour? I'm not even sure what _should_ happen. 😅

Reply:
we can add tests for this, but the way I see it, `fetchQuery` would just pick up the promise that's already running and not start a new fetch. Of course, if you do `queryClient.refetchQueries` and target this key, it would _cancel_ the ongoing fetch per default. Not sure what would happen then, cancel should silently rollback to the previous (pending) state, so maybe it would hang forever.

Definitely interesting case that's worth a couple of tests 👍